packages/dev/core/src/scene.ts

```diff
@@ -2895,18 +2895,11 @@ export class Scene extends AbstractScene implements IAnimatable, IClipPlanesHold
     }
 
     private _getMaterial(allowMultiMaterials: boolean, predicate: (m: Material) => boolean): Nullable<Material> {
-        for (let index = 0; index < this.materials.length; index++) {
-            const material = this.materials[index];
-            if (predicate(material)) {
-                return material;
-            }
-        }
-        if (allowMultiMaterials) {
-            for (let index = 0; index < this.multiMaterials.length; index++) {
-                const material = this.multiMaterials[index];
-                if (predicate(material)) {
-                    return material;
-                }
+        const targetMaterials = allowMultiMaterials ? this.multiMaterials : this.materials;

```

Comment:
this won t work as it is not a OR situation. It is first looking in one, than the other one.

We can not concat here to prevent extra GC and generatops would require more tslib support unfortunately :-(

Reply:
Oops my mistake =)
So, I added new test cases that called as "should looking for material first before looking for multiMaterial". Check it again, please!